### PR TITLE
add retry to docker pull 

### PR DESCRIPTION
**Pull request by bogyo210**

### CODE REVIEW

File: sem-service [hunk 1]
```diff
@@ -3,6 +3,14 @@
 # Misc
 DATE_FORMAT='%H:%M %d/%m/%Y'
 
+service::pull_image_retry() {
+  counter=1 
+  local image_name=$2
+  while ! docker pull $image_name && [ $counter -lt 3 ] ;do 
+    counter=$((counter+1));
```
Comment (shiroyasha) [inline sem-service:10]: Add a bit of sleep between retries. For example, 1 second.

**Comment by shiroyasha** [comment]
Looks good. 

Please also run it throught shell-check. If OK, shipit.

**Comment by bogyo210** [comment]
Not sure about the last line, the rest passed:
https://semaphore.semaphoreci.com/jobs/d651e6bd-14c3-4b96-93ad-5cfcffc82ced

**Comment by shiroyasha** [comment]
@bogyo210 thanks for adding it. My idea with this was:

- It is not feasable to fix every problem atm
- We can still use shell-check to limit the number of newly introduced problems.

For the cache script, it is 154 errors atm. I set it incorrectly to 152. Please modify this.

**Comment by shiroyasha** [comment]
Screen Shot 2020-06-19 at 2 56 22 PM

However, now when I check it, every number is off... Damn... Any idea how to achive this?

I want to set an upper limit for errors to make sure that we are not introducing new ones.

**Comment by bogyo210** [comment]
merged to nb/shellcheck, pr will come from that

**Pull request closed without merging**